wfl-web-api/wfl/Controllers/RestaurantsController.cs

```diff
@@ -49,11 +50,21 @@ public RestaurantsController(IDBManager<Restaurant> dbManager)
 
         [Route("")]
         [HttpGet]
-        public async Task<IHttpActionResult> GetAsync()
+        public async Task<IHttpActionResult> GetAsync([FromUri] NameSearchCriteria criteria)
         {
+            IEnumerable<Restaurant> restaurants;
+            if (String.IsNullOrEmpty(criteria?.Name))
```

Comment:
If we were to add more criteria, would we just expand this if statement? May be useful to expose a `bool HasCriteria()` on the criteria base class, or don't add this check in the controller at all and use it when creating the mongo query.